sklearn/utils/validation.py

```diff
@@ -374,6 +374,9 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
 
         if ensure_2d:
             if array.ndim == 1:
+                if ensure_min_samples >= 2:
+                    raise ValueError("%s expects at least 2 samples provided " 
+                    "in a 2 dimensional array-like input")
```

Comment:
The indentation level does not follow PEP8, use a linter such as https://pypi.python.org/pypi/pep8 or better: https://pypi.python.org/pypi/flake8 to spot such issues.